tools/individual-pkg-gen/copy-files-removing-deps.ts

```diff
@@ -0,0 +1,217 @@
+                break;
+              default:
+                if (alphaPackages[v1DevDependency]) {
+                  alphaDevDependencies[alphaPackages[v1DevDependency]] = pkg.version;
+                } else if (!v1DevDependency.startsWith('@aws-cdk/')) {
+                  devDependencies[v1DevDependency] = packageJson.devDependencies[v1DevDependency];
+                }
+            }
+          }
+          const finalPackageJson = { ...packageJson };
+          // we save the devDependencies in a temporary _package.json
+          finalPackageJson.devDependencies = {
+            ...devDependencies,
+            ...constructsAndCdkLibDevDeps,
+            ...alphaDevDependencies,
+            ...alphaDependencies,
+          };
+          packageJson.devDependencies = {
+            ...alphaDevDependencies,
+            ...alphaDependencies,
+          };
+
+          // peer dependencies
+          finalPackageJson.peerDependencies = {
+            ...(Object.entries(alphaDependencies)
+              // for other alpha dependencies, we need to depend on exact versions
+              // (because of the braking changes between them)
+              .reduce((acc, [depName, depVersion]) => {
+                acc[depName] = depVersion;
+                return acc;
+              }, {} as { [dep: string]: string })),
+            ...(Object.entries(constructsAndCdkLibDevDeps)
+              .reduce((acc, [depName, depVersion]) => {
+                acc[depName] = `${depVersion.startsWith('^') ? '' : '^'}${depVersion}`;
+                return acc;
+              }, {} as { [dep: string]: string })),
+          };
+          packageJson.peerDependencies = undefined;
+
+          fileProcessed = true;
+          fs.writeJsonSync(destination, packageJson, { spaces: 2 });
+          fs.writeJsonSync(path.join(destDir, '_package.json'), finalPackageJson, { spaces: 2 });
+        }
+      } else if (sourceFileName === '.gitignore') {
+        // ignore everything, otherwise there are uncommitted files present in testing,
+        // because the module's .gitignore file has entries like !.eslintrc.js
+        fileProcessed = true;
+        const gitIgnoreContents = fs.readFileSync(source);
+        fs.outputFileSync(destination, Buffer.concat([gitIgnoreContents, Buffer.from('\n*\n')]));
+      } else if (sourceFileName === '.eslintrc.js') {
+        // Change the default configuration of the import/no-extraneous-dependencies rule
+        // (as the unstable packages don't use direct dependencies,
+        // but instead a combination of devDependencies + peerDependencies)
+        fileProcessed = true;
+        const esLintRcContents = fs.readFileSync(source);
+        fs.outputFileSync(destination, Buffer.concat([esLintRcContents,
+          Buffer.from("\nbaseConfig.rules['import/no-extraneous-dependencies'] = ['error', " +
+              '{ devDependencies: true, peerDependencies: true } ];\n')]));
```

Comment:
something about this still feels off.

In the AWS CDK CLI - `nock` is a `devDependency` - https://github.com/aws/aws-cdk/blob/master/packages/aws-cdk/package.json#L64
This rule doesn't complain here - https://github.com/aws/aws-cdk/blob/7966f8d48c4bff26beb22856d289f9d0c7e7081d/packages/aws-cdk/test/api/fake-sts.ts#L1

Reply:
Right, because `true` means "allow importing from that set of dependencies". Which is exactly what we need for the unstable modules.

Reply:
As for the case you alluded to above, [we explicitly allow tests to import from `"devDependencies"`](https://github.com/aws/aws-cdk/blob/f3bf628d0af71011a7b7f1ca3dca2ce036011329/packages/awslint/.eslintrc.js#L63-L66).